### PR TITLE
[MODCON-6]-added endpoint for inserting tenant.

### PR DESCRIPTION
**Pull request by SinghAdes**
## Purpose

[MODCON-6](https://issues.folio.org/browse/MODCON-6)
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -15,6 +15,16 @@
...
           ],
           "modulePermissions": []
+        },
+        {
+          "methods": [
+            "POST"
+          ],
+          "pathPattern": "/consortia/tenants",
+          "permissionsRequired": [
+            "consortia.tenants.post"
```
Comment (SerhiiNosko) [inline descriptors/ModuleDescriptor-template.json:25]: We should follow the same naming conventions, in this case we should use this name: consortia.tenants.item.post

File: src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java [hunk 2]
```diff
@@ -28,6 +28,15 @@ public TenantCollection get(Integer offset, Integer limit) {
     return result;
   }
 
+  @Transactional
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java:31]: We just saving 1 entity to DB, we can do it without transaction

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (SinghAdes) [reply descriptors/ModuleDescriptor-template.json]: fixed

File: src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java (hunk 2 above)
Comment (SinghAdes) [reply src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java]: fixed

File: src/test/java/org/folio/consortia/domain/EntityTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import org.folio.consortia.domain.entity.TenantEntity;
+import org.folio.consortia.domain.entity.UserTenantEntity;
+import org.junit.jupiter.api.Test;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+class EntityTest {
```
Comment (Abdulkhakimov) [inline src/test/java/org/folio/consortia/domain/EntityTest.java:12]: Do we really need tests to test entities?

**Review by Abdulkhakimov — APPROVED** [review]

**Comment by Abdulkhakimov** [comment]
@SinghAdes please check sonar warning. There is code smell here

File: src/test/java/org/folio/consortia/domain/EntityTest.java (hunk 3 above)
Comment (SinghAdes) [reply src/test/java/org/folio/consortia/domain/EntityTest.java:12]: Not really needed but just to be on safer side.

**Comment by SinghAdes** [comment]
> @SinghAdes please check sonar warning. There is code smell here

fixed

**Pull request closed without merging**